### ISSUE TITLE
bootstrapper can leave the VE in broken state

### ISSUE THREAD

**Issue by andre-merzky** [radical-cybertools/radical.pilot#911]
or at least that is I think what happened.  I found that runs on stampede started failing, and started to look into it.  The bootstrapper's call to `radicalpilot-version` complained about pyzmq not being importable, despite the fact that the `pip install radical.pilot` output above indicated that pyzmq installation existed in the VE and no install was needed.  It turned out that the egg-files indeed existed, but no actual modules -- and pip seems to only check the egg existence.   I assume that the VE gut butchered somehow by terminating the bootstrapper at some unfortunate point in time, but do not know if that was indeed the cause.  Uninstalling the remains of pyzmq fixed the issue.

Bottom line: if ve seems to install ok, but fails at runtime, it might be worth to re-create it.  I think Mark suggested something similar in the past...

**Comment by mturilli** [comment]
In a 'similar' situation I found a combination of the following useful: ```--force-reinstall```, ```--ignore-installed```, ```--upgrade``` depending on whether you are upgrading. Probably they do not apply to your case but just in case.

**Comment by andre-merzky** [comment]
They would pessimise the default bootstrapping -- but for a repair attempt, that looks useful!  Thanks! :)

**Comment by andre-merzky** [comment]
closed as  duplicate of #555 555
